benchmark/src/benchmark.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/benchmark
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as byline from 'byline';
+import {ChildProcess, spawn} from 'child_process';
+import * as pEvent from 'p-event';
+import {Autocannon, EndpointStats} from './autocannon';
+import {Client} from './client';
+import {scenarios} from './scenarios';
+
+export interface Scenario {
+  setup(client: Client): Promise<void>;
+  execute(autocannon: Autocannon): Promise<EndpointStats>;
+}
+
+export type ScenarioFactory = new () => Scenario;
+
+export interface Options {
+  /**
+   * How long to run the benchmark - time in seconds.
+   * Default: 30 seconds.
+   */
+  duration: number;
+
+  /**
+   * Optional callback to log progress.
+   */
+  logger: (title: string, stats: EndpointStats) => void;
+}
+
+export interface Result {
+  [scenario: string]: EndpointStats;
+}
+
+export class Benchmark {
+  private options: Options;
+  private worker: ChildProcess;
+  private url: string;
+
+  constructor(options?: Partial<Options>) {
+    this.options = Object.assign(
+      {
+        duration: 30 /* seconds */,
```

Comment:
should we up the number of concurrent connections from a default of 10 to 50 / 100?

Reply:
To be honest, my knowledge of performance testing of HTTP is limited and I cannot asses possible ramifications of changing the number of concurrent connections from 10 to 50 or 100. I assume the authors of autocannon have choosen a default value that should work well for majority of users.

What is your reasoning for changing this config option?